api/envoy/api/v2/cds.proto

```diff
@@ -539,6 +539,10 @@ message Cluster {
       // * :ref:`runtime values <config_cluster_manager_cluster_runtime_zone_routing>`.
       // * :ref:`Zone aware routing support <arch_overview_load_balancing_zone_aware_routing>`.
       google.protobuf.UInt64Value min_cluster_size = 2;
+
+      // If set to true, Envoy will not consider any hosts when the cluster is in panic mode.
+      // Instead, the cluster will fail all requests as if all hosts are unhealthy.
+      bool disable_cluster_on_panic = 3;
```

Comment:
Why not set panic threshold to 0% to disable as per https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/panic_threshold.html?highlight=panic? Is there some other advantage to being in panic mode but not doing panic behavior?

Reply:
The idea behind this patch is not just to disable sending requests to hosts once they've been detected as unhealthy, but to also stop sending requests to _healthy_ hosts once the panic threshold has been reached in order to avoid potentially overwhelming a failing service.